src/main/java/org/opensrp/config/RabbitMQConfig.java

```diff
@@ -0,0 +1,68 @@
+package org.opensrp.config;
+
+import org.opensrp.queue.RabbitMQReceiver;
+import org.springframework.amqp.core.*;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.amqp.rabbit.listener.MessageListenerContainer;
+import org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainer;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class RabbitMQConfig {
+
+
+	@Value("${rabbitmq.queue}")
+	String queueName;
+
+	@Value("${rabbitmq.exchange}")
+	String exchange;
+
+	@Value("${rabbitmq.routingkey}")
+	private String routingkey;
+
+	@Bean
+	Queue queue() {
+		return new Queue(queueName, false);
+	}
+
+	@Bean
+	DirectExchange exchange() {
+		return new DirectExchange(exchange);
+	}
+
+	@Bean
+	Binding binding(Queue queue, DirectExchange exchange) {
+		return BindingBuilder.bind(queue).to(exchange).with(routingkey);
+	}
+
+	@Bean
+	public MessageConverter jsonMessageConverter() {
+		return new Jackson2JsonMessageConverter();
+	}
+
+
+	@Bean
+	public AmqpTemplate rabbitTemplate(ConnectionFactory connectionFactory) {
+		final RabbitTemplate rabbitTemplate = new RabbitTemplate(connectionFactory);
+		rabbitTemplate.convertAndSend(jsonMessageConverter());
+		rabbitTemplate.setMessageConverter(jsonMessageConverter());
+		return rabbitTemplate;
+	}
+
+
+	//create MessageListenerContainer using default connection factory
+	@Bean
+	MessageListenerContainer messageListenerContainer(ConnectionFactory connectionFactory ) {
+		SimpleMessageListenerContainer simpleMessageListenerContainer = new SimpleMessageListenerContainer();
+		simpleMessageListenerContainer.setConnectionFactory(connectionFactory);
+		simpleMessageListenerContainer.setQueues(queue());
+		simpleMessageListenerContainer.setMessageListener(new RabbitMQReceiver());
```

Comment:
Do we want a Bean of RabbitMQReceiver?

Reply:
Yes, I think we need to define the default message listener.